snippets/SharedHandle.hpp

```diff
@@ -221,49 +122,54 @@ class SharedHandleBase
     return &m_handle;
   }
 
-  size_t add_ref() VULKAN_HPP_NOEXCEPT
+  void reset() VULKAN_HPP_NOEXCEPT
   {
     if ( !m_handle )
-      return 0;
-    return m_control.add_ref();
+      return;
+
+    auto refs = m_control.use_count();
+    if ( refs == 1 )
+      static_cast<SharedHandle<HandleType> *>( this )->internalDestroy();
+    m_control.reset();
+    m_handle = nullptr;
   }
 
-  size_t release() VULKAN_HPP_NOEXCEPT
+  template <typename T = HandleType>
+  typename std::enable_if<has_parent<T>, ParentType>::type getParent() const VULKAN_HPP_NOEXCEPT
```

Comment:
I would not change the level of abstraction here. Instead, I would provide just one `getParent()`, instead of two, but it should return a `SharedHandle<ParentType> const &`!